hwbotprime-desktop/src/main/java/org/hwbot/bench/ui/swing/BenchSwingUI.java

```diff
@@ -275,26 +277,42 @@ public void actionPerformed(java.awt.event.ActionEvent evt) {
     }// </editor-fold>//GEN-END:initComponents
 
     private void startButtonStabilityActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FIRST:event_startButtonStabilityActionPerformed
-        // TODO add your handling code here:
+        isStability = true;
+        System.out.println("started stability benchmark this may take a while");
+        this.startButton.setEnabled(false);
+        this.startButtonStability.setEnabled(false);
+        this.saveButton.setEnabled(false);
+        this.compareButton.setEnabled(false);
+
+        int stabilityBenchmarkTimeSec = 3600;
+        this.benchService.benchmark(stabilityBenchmarkTimeSec);
+
     }// GEN-LAST:event_startButtonStabilityActionPerformed
 
     private void compareButtonActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FIRST:event_compareButtonActionPerformed
         benchService.submit();
     }// GEN-LAST:event_compareButtonActionPerformed
 
     private void startButtonActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FIRST:event_jButton1ActionPerformed
+        isStability = false;
         this.startButton.setEnabled(false);
         this.compareButton.setEnabled(false);
         this.saveButton.setEnabled(false);
         this.startButtonStability.setEnabled(false);
-        this.benchService.benchmark();
+        this.benchService.benchmark(null);
     }// GEN-LAST:event_jButton1ActionPerformed
 
     private void saveButtonActionPerformed(ActionEvent evt) {
         //set filechooser defaults
         FileNameExtensionFilter extensions = new FileNameExtensionFilter("HWBOT Scorefiles", "hwbot");
         fc.setFileFilter(extensions);
-        fc.setSelectedFile(new File(score.getText()));
+        if (isStability){
+            fc.setSelectedFile(new File("stability benchmark: " + score.getText()));
+        }else{
+            fc.setSelectedFile(new File("quick benchmark: " + score.getText()));
+        }
+
+
```

Comment:
the file name looks a bit odd, change to stability-<score>.hwbot and quick-<score>.hwbot